File: DATHelper/app/src/main/java/com/kishanprao/datHelper/base/stateMachine/Action.kt
package com.kishanprao.datHelper.base.stateMachine

sealed class Action {
    class OnConnect(val serverAddress: String) : Action()
    class OnDisconnect : Action()
    class OnOperation : Action()
    class OnOperationCompleted : Action()
    class OnFailure(val code: Int) : Action()
    class OnFailureHandled : Action()

    override fun toString(): String {
        return "[ACTION:${this::class.java.simpleName}]"
    }
}

File: Dragon Android Transfer/Dragon Android Transfer/Base/Wireless/StateMachine/Action.swift
//
//  Action.swift
//  DragonSocket
//
//  Created by Kishan P Rao on 08/11/18.
//  Copyright © 2018 Kishan P Rao. All rights reserved.
//

import Foundation

enum Action {
    case OnConnect()
    case OnDisconnect()
    case OnOperation(OperationData)
    case OnOperationCompleted()
    case OnFailure(Int)
    case OnFailureHandled()
}
